File: chrome/browser/resources/tracing/test_utils.js
// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview Helper functions for use in tracing tests.
 */


/**
 * goog.testing.assertion's assertEquals tweaked to do equality-to-a-constant.
 * @param {*} a First value.
 * @param {*} b Second value.
 */
function assertAlmostEquals(a, b) {
  _validateArguments(2, arguments);
  var var1 = nonCommentArg(1, 2, arguments);
  var var2 = nonCommentArg(2, 2, arguments);
  _assert(commentArg(2, arguments), Math.abs(var1 - var2) < 0.00001,
          'Expected ' + _displayStringForValue(var1) + ' but was ' +
          _displayStringForValue(var2));
}

cr.define('test_utils', function() {
  function getAsync(url, cb) {
    var req = new XMLHttpRequest();
    req.open('GET', url, true);
    req.onreadystatechange = function(aEvt) {
      if (req.readyState == 4) {
        window.setTimeout(function() {
          if (req.status == 200) {
            cb(req.responseText);
          } else {
            console.log('Failed to load ' + url);
          }
        }, 0);
      }
    };
    req.send(null);
  }
  return {
    getAsync: getAsync
  };
});

File: pdf/test/test_utils.cc
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "pdf/test/test_utils.h"

#include "ppapi/cpp/rect.h"
#include "testing/gtest/include/gtest/gtest.h"

namespace chrome_pdf {

void CompareRect(const pp::Rect& expected_rect, const pp::Rect& given_rect) {
  EXPECT_EQ(expected_rect.x(), given_rect.x());
  EXPECT_EQ(expected_rect.y(), given_rect.y());
  EXPECT_EQ(expected_rect.width(), given_rect.width());
  EXPECT_EQ(expected_rect.height(), given_rect.height());
}

void CompareSize(const pp::Size& expected_size, const pp::Size& given_size) {
  EXPECT_EQ(expected_size.width(), given_size.width());
  EXPECT_EQ(expected_size.height(), given_size.height());
}

}  // namespace chrome_pdf